### PR TITLE
Use lazy loading instead of one-off loading

### PR DESCRIPTION
**Pull request by chengfx**
Modify the load method in building dictionary and encoding data in order to avoiding memory overflow

### CODE REVIEW

**Comment by chengfx** [comment]
> thanks Feixiang for the request, this is very important feature. Have we tested a large scale data, such as knowledge disllation using like 1000 million data?

Hi, @ljshou . I have tested Query Binary Classification task using 35 million data( The size of file is about 1.5G) In my local PC( Memory 32G) already. It can run  normally, including building dictionary, encoding raw data and training.
In building dictionary phase, there will not be memory overflow if the size of word universe is less than that of memory.  I tested it using 200 million data(the size of file is about 8.4G). The process just need about 10G memory(***majorly used by word universe***). There is a function parameter ***chunk_size*** which can control the number of line loaded by process.  I think it could be added as the config parameter later.
In encoding phase, we make an assumption that all encoded data could be loaded in memory. Otherwise, there will be a out of memory exception.

For 1000 million data, I think current version could build dictionary, but will fail at encoding raw data. :smile:

File: problem.py [hunk 1]
```diff
@@ -285,6 +300,7 @@ def build(self, training_data_path, file_columns, input_types, file_with_col_hea
 
             assert loaded_emb_dim == word_emb_dim, "The dimension of defined word embedding is inconsistent with the pretrained embedding provided!"
 
+            logging.info("constrct embedding table")
```
Comment (woailaosang) [inline problem.py]: constrct -> construct ?

Comment (chengfx) [reply problem.py]: thanks, Zhijie, done

**Comment by ljshou** [comment]
thanks feixiang

**Pull request merged**